src/druglab/pipe/blocks/featurize.py

```
@@ -0,0 +1,181 @@
+            # The Gobbi fingerprint length varies; we fold/truncate to n_bits.
+            fp_len = fp.GetNumBits()
+            arr_full = np.zeros(fp_len, dtype=np.int8)
+            DataStructs.ConvertToNumpyArray(fp, arr_full)
+ 
+            if fp_len >= self.n_bits:
+                return arr_full[: self.n_bits]
+            else:
+                out = np.zeros(self.n_bits, dtype=np.int8)
+                out[:fp_len] = arr_full
+                return out
+        except Exception:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)